ci/install_format_tools.sh

```diff
@@ -2,7 +2,8 @@
 
 set -e
 
-apt install -y clang-format-8 python3-pip git curl
+apt-get update
+apt-get install -y clang-format-11 python3-pip git curl
```

Comment:
Nit: [format.sh](https://github.com/open-telemetry/opentelemetry-cpp/blob/cae3511915ab907ab1f30c620b0efb2634fa5862/tools/format.sh#L27) also needs to be modified to ensure right version is picked in case there are multiple versions installed.

Reply:
So far I was not able to find the package to install... Some help would be appreciated

Reply:
Which OS  ?  I can see multiple clang-format versions for ubuntu-latest in default repo:
```
root@e4c04138cab4:/# apt-cache search clang-format
arcanist-clang-format-linter - clang-format linter for Arcanist
clang-format - Tool to format C/C++/Obj-C code
clang-format-10 - Tool to format C/C++/Obj-C code
clang-format-6.0 - Tool to format C/C++/Obj-C code
clang-format-7 - Tool to format C/C++/Obj-C code
clang-format-8 - Tool to format C/C++/Obj-C code
clang-format-9 - Tool to format C/C++/Obj-C code
clang-format-11 - Tool to format C/C++/Obj-C code
``

Reply:
See the github action failing message:

```
Reading state information...
E: Unable to locate package clang-format-11
```

Reply:
Ok. Missed out the failure.
`ubuntu-18.04` and `ubuntu-20.04` virtual machines in github doesn't bring clang-format-11 :
https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md :  `Clang 8.0.1, 9.0.1, 10.0.0`
https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-README.md : `Clang 6.0.0, 8.0.0, 9.0.0`

Reply:
option I can see is to either use to `llvm` repo to install the right package ( https://apt.llvm.org/), or switch to macOS for formatting as it brings the clang-11 version. cc @maxgolov

Reply:
I'd suggest we use the one available in apt-cache with highest version (clang-format-10?), as we usually need run clang-format locally so no need to complicate it with building from source or install binary package.

Reply:
Agree

Reply:
I think the best long term option is to have a Docker container with all this formating/linting tools that everyone can use

Reply:
See https://github.com/open-telemetry/build-tools/pull/28 for the docker image and instructions to run

Reply:
Please @lalitb @ThomsonTan @maxgolov review that PR :)